src/pip/_internal/network/session.py

```diff
@@ -422,7 +422,8 @@ def add_trusted_host(
 
         host_port = parse_netloc(host)
         if host_port not in self.pip_trusted_origins:
-            self.pip_trusted_origins.append(host_port)
+            assert host_port[0] is not None
```

Comment:
Could we not put this check outside the conditional, and make it into a “real” error (“trusted host url must include a host part”)? To put it another way, this comes from user entered data (I believe) so let’s validate that data at source.